src/Altinn.App.Core/Models/AppIdentifier.cs

```diff
@@ -22,8 +24,82 @@ public class AppIdentifier
         /// <param name="app">The app name.</param>
         public AppIdentifier(string org, string app)
         {
+            if (string.IsNullOrEmpty(org))
+            {
+                throw new ArgumentNullException(nameof(org));
+            }
+
+            if (string.IsNullOrEmpty(app))
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
             Org = org;
             App = app;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppIdentifier"/> class.
+        /// </summary>
+        /// <param name="id">Application id on the form org/app</param>
+        public AppIdentifier(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (id.ContainsMoreThanOne('/'))
```

Comment:
Would it be better to implement a method string.ContainsExactlyOne ?

Reply:
Yep:)